internal/kubernetes/util.go

```diff
@@ -48,3 +53,38 @@ func NewEventRecorder(c kubernetes.Interface) record.EventRecorder {
 	b.StartRecordingToSink(&typedcore.EventSinkImpl{Interface: typedcore.New(c.CoreV1().RESTClient()).Events("")})
 	return b.NewRecorder(scheme.Scheme, core.EventSource{Component: Component})
 }
+
+func Until(f func() error, retryPeriod, timeout time.Duration) error {
```

Comment:
Taking my time with this guy. A comment would helpful. I'm curious also why we didn't use the apimachinery `wait.Until` It looks like the intention is:

- call f until it succeeds or until the timeout is reached.
- retry every retryPeriod as long as f() does not succeed.

It looks like `wait.Until` offers a "stopChan" parameter. Could we not, for example:

```
ctx, cancel := context.WithTimeout(context.TODO(), timeoutMs)
defer cancel()
wait.Until(f, period, ctx.Done())
```

It looks like we largely just want to offer a different API for the caller here, which I can appreciate.

Reply:
Thanks to your comment and after giving a better look to the package `wait`, I think I am going to remove that code and use `wait.PollImmediateUntil`
Because what I need here is a retry with timeout. The name `Until` was not a good choice.

Reply:
e4ff89daab37b96b20a3a92675cba60afa44af14